README.md

```diff
@@ -1,45 +1,77 @@
+
+- [Enterprise Contract Documentation](https://enterprisecontract.dev/docs/ec-cli/main/ec_validate_image.html#_synopsis)
+- [Red Hat AppStudio Documentation](https://redhat-appstudio.github.io/book/book/enterprise-contract.html#:~:text=EC%20CLI,or%20violations%20produced)
+
+
+## Environment Variables 
+
+To use this action, please configure the following environment variables in your workflow based on the desired authentication method.
+
+### Using Long-Lived Public-Key Authentication
+
+The Long-Lived public-key method offers a straightforward way to integrate with Sigstore. Below are the variables needed:
 
-To run this action, you will need to add the following variables to your workflow.
 | Name          | Description                                                                                      | Example                                     |
 |---------------|--------------------------------------------------------------------------------------------------|---------------------------------------------|
-| Public Key    | The public key for verifying signatures.                                                | `"your_public_key_goes_here"`                 |
-| Policy        | The location of the policy.yaml config file to be used when running Enterprise Contract. A list of standard configs can be found at [here](https://github.com/enterprise-contract/config).  | `"github.com/enterprise-contract/config//slsa3"` |
-| Image         | Image that is built.                                                                            | `"quay.io/redhat-appstudio/ec-golden-image:latest"` |
+| Public Key    | The public key for verifying signatures.                                                | `your_public_key_goes_here`                 |
+| Policy        | The location of the policy.yaml config file to be used when running Enterprise Contract. A list of standard configs can be found at [here](https://github.com/enterprise-contract/config).  | `github.com/enterprise-contract/config//slsa3` |
+| Image         | Image that is built.                                                                            | `quay.io/redhat-appstudio/ec-golden-image:latest` |
 
+### Using Identity-Based Short-Lived (Keyless) Authentication
 
+This authentication approach is the most robust, as Sigstore authentication is based on the signer's identity rather than a key. The following variables are needed:
 
+| Name          | Description                                                                                      | Example                                     |
+|---------------|--------------------------------------------------------------------------------------------------|---------------------------------------------|
+| Identity      | Identity to verify the certificate linked to the image.                                          | `https:\/\/github\.com\/(slsa-framework\/slsa-github-generator\|lcarva\/festoji)\/`                 |
```

Comment:
The example value is a regex - I think the description needs to mention that. Question: Is it always treated as a regex? Is an exact match with a plain string possible?

Reply:
We could add support for '--certificate-identity' if we wanted to.

Reply:
A simple change that could work:

```suggestion
| Identity      | A regexp string used to match the certificate identity linked to the image.                                          | `https:\/\/github\.com\/(slsa-framework\/slsa-github-generator\|lcarva\/festoji)\/`                 |
```